controllers/chat/sendAnonymousMessage.js

```diff
@@ -56,13 +56,14 @@ const sendAnonymousMessage = async (req, res) => {
     if (
       req.user.is_anonymous &&
       createdChannel.channel.anon_user_info_emoji_name &&
-      client.user.name !== `Anonymous ${createdChannel.channel.anon_user_info_emoji_name}`
+      client.user.name !==
+        `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`
     ) {
       await client.upsertUser({
         id: req.userId,
-        name: `Anonymous ${createdChannel.channel.anon_user_info_emoji_name}`,
+        name: `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`,
         image: createdChannel.channel.anon_user_info_emoji_code,
-        username: `Anonymous ${createdChannel.channel.anon_user_info_emoji_name}`
+        username: `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes look good, but there's a potential issue. If the `anon_user_info_color_name` or `anon_user_info_emoji_name` is null or undefined, it could lead to an unexpected username. It would be better to add checks for these properties before using them.

```diff
- if (
-   req.user.is_anonymous &&
-   createdChannel.channel.anon_user_info_emoji_name &&
-   client.user.name !== `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`
- ) {
+ if (
+   req.user.is_anonymous &&
+   createdChannel.channel.anon_user_info_emoji_name &&
+   createdChannel.channel.anon_user_info_color_name &&
+   client.user.name !== `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`
+ ) {
    await client.upsertUser({
      id: req.userId,
      name: `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`,
      image: createdChannel.channel.anon_user_info_emoji_code,
      username: `${createdChannel.channel.anon_user_info_color_name} ${createdChannel.channel.anon_user_info_emoji_name}`
    });
  }
```
This way, we ensure that both `anon_user_info_color_name` and `anon_user_info_emoji_name` are not null or undefined before using them.


<!-- This is an auto-generated comment by OSS CodeRabbit -->